.github/workflows/build-test-analyze.yml

```diff
@@ -69,6 +69,71 @@ jobs:
           flags: unittests
           name: codecov-coverage
           token: ${{ secrets.CODECOV_TOKEN }}
+  test:
```

Comment:
Since we already run unit tests and upload the unit tests coverage as part of `buildtestandanalyze` (which runs on pull requests), would it make sense to merge the steps of a single matrix configuration in `buildtestandanalyze` and upload coverage once? We can then think of executing all matrix configurations of this current job when merging to main. Also, I noticed that there have been [updates](https://github.com/StanfordSpezi/SpeziKt/pull/30) in the workflows, we are now also deploying Dokka documentation on every PR. I missed the review from #30, should we move `dokka` job back to main workflow?

Reply:
I tried to look into making this a single matrix but it gets tricky with the fact that we need a different setup for unit tests and Android tests + we only really need variations on the devices + SDK levels for the connected tests. I would therefore suggest to keep it that way for now, the reference from the main branch will execute:
1. All unit tests + upload the code coverage there
2. All connected tests in a test matrix of devices + android versions that we can expand on if needed.

Codecov then merges these reports in the backend anyways so there is no harm if we upload them from multiple different jobs and elements in the Action as long as all of them are available after all tests are run and that seems to be the case.

Regarding docs: I moved them here in the main build action as they were failing on the main branch due to a syntax error in the MD files. To avoid this, we build the docs with every PR, but we don't deploy them (if statement in the last step). It is basically a "dry run" if the docs building and export works but the actual step to deploy them to GitHub pages only happens if the ref is the main branch and therefore only happens on merged PRs on the main branch. This ensures that we can fix docs errors or occurrences where that might fail before we even merge a PR and don't run into a regression like the one we had the last time again. Or at least let's hope so 😄